sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/FileFormatWriter.scala

```diff
@@ -117,7 +117,7 @@ object FileFormatWriter extends Logging {
     job.setOutputValueClass(classOf[InternalRow])
     FileOutputFormat.setOutputPath(job, new Path(outputSpec.outputPath))
 
-    val allColumns = plan.output
+    val allColumns = queryExecution.logical.output
```

Comment:
I think it'd be good to leave a comment that we should not use optimized output here in case it will be changed in the future.

Reply:
Btw, shall we use `queryExecution.analyzed.output`?